backend/src/org/commcare/data/xml/DataModelPullParser.java

```diff
@@ -119,14 +119,11 @@ private void parseBlock(String root) throws InvalidStructureException, IOExcepti
             }
 
             String name = parser.getName();
-            String namespace = parser.getNamespace();
-
-            int depth = parser.getDepth();
             if (name == null) {
                 continue;
             }
 
-            TransactionParser transaction = factory.getParser(name, namespace, parser);
```

Comment:
Since this is the only place that getParser is actually called, and it is always using the name and namespace from the 'parser' object, I've changed the signature of getParser to only take in the parser as an argument.